lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -522,6 +520,32 @@ function dosomething_reportback_form_alter(&$form, &$form_state, $form_id) {
   }
 }
 
+/**
+ * Callback for /rbf/FID page.
+ */
+function dosomething_reportback_permalink_view($args) {
+  $rbid = $args[0];
+  $fid = $_GET['fid'];
+
+  if (variable_get('rogue_collection', FALSE)) {
+    return dosomething_reportback_view_rogue($rbid, $fid);
+  }
+
+  return dosomething_reportback_view_entity(entity_load('reportback', $rbid));
+}
+
+/**
+ * Callback for /rbf/FID page, when Rogue is enabled.
+ */
+function dosomething_reportback_view_rogue($rbid, $fid) {
+    // @TODO: Request signup from Rogue so we can display it on permalink page.
+    // $rogue_signup = dosomething_rogue_client()->get('v2/activity')
```

Comment:
yes @sbsmith86! Right now, there are no endpoints in the Rogue API that return information for a specific signup. We'll have to build this out (and all the work to build out the Rogue permalink page that better mimics what is going on is a commit in this PR) and once it's finished, should be pretty easy to do full permalink page!